spec/play_spec.rb

```diff
@@ -5,29 +5,68 @@
 require './lib/player_choice'
 require './lib/play'
 require './spec/player_choice_spec'
+require './mock/test_ui'
+require './mock/test_player_choice'
+require './mock/test_board'
 
 describe 'Play' do
   before(:each) do
-    @end_of_game = GameOver.new
+    @test_ui = TestUI.new
+    @end_of_game = GameOver.new(output: @test_ui)
     @board = Board.new
     @cpu = CPUChoice.new(@end_of_game)
-    @test_ui = TestUI.new
   end
 
-  describe '.player_picks_a_spot' do
-    it 'Player chooses 5 and the board is marked with their choice' do
-      input = StringIO.new('5')
-      @player = PlayerChoice.new(input: input, output: @test_ui)
-      @play = Play.new(@board, @cpu, @player, @end_of_game)
-      @play.player_picks_a_spot
-      expect(@board.board).to eq(%w[1 2 3 4 X 6 7 8 9])
+  describe '.run' do
```

Comment:
I think you're missing a unit test for a scenario. So far you have a player making an invalid move, making a move on a spot already taken, a win scenario, and a draw scenario. Is the unit test(s) for the losing scenario covered somewhere else?

Reply:
No, I didn't create a losing message. I only created a message that returns which marker won. I could see though how it would make sense to have a losing message in a single-player game.